lib/events/api.go

```diff
@@ -640,6 +640,27 @@ const (
 	// SAMLIdPServiceProviderDeleteAllEvent is emitted when all service providers have been deleted.
 	SAMLIdPServiceProviderDeleteAllEvent = "saml.idp.service.provider.delete_all"
 
+	// OktaGroupsUpdate event is emitted when the groups synced from Okta have been updated.
+	OktaGroupsUpdateEvent = "okta.groups.update"
+
+	// OktaApplicationsUpdateEvent is emitted when the applications synced from Okta have been updated.
+	OktaApplicationsUpdateEvent = "okta.applications.update"
+
+	// OktaSyncFailureEvent is emitted when the Okta synchronization fails.
+	OktaSyncFailureEvent = "okta.sync.failure"
+
+	// OktaAssignmentProcessSuccessEvent is emitted when an assignment is successfully processed.
+	OktaAssignmentProcessSuccessEvent = "okta.assignment.process"
+
+	// OktaAssignmentProcessFailureEvent is emitted when an assignment is unsuccessfully processed.
+	OktaAssignmentProcessFailureEvent = "okta.assignment.process.failure"
+
+	// OktaAssignmentCleanupSuccessEvent is emitted when an assignment is successfully cleaned up.
+	OktaAssignmentCleanupSuccessEvent = "okta.assignment.cleanup"
+
+	// OktaAssignmentCleanupFailureEvent is emitted when an assignment is unsuccessfully cleaned up.
+	OktaAssignmentCleanupFailureEvent = "okta.assignment.cleanup.failure"
```

Comment:
You're right. There is prior art for the failure as a separate event:
```
	DatabaseSessionQueryFailedEvent = "db.session.query.failed"
```
But that seems like an error. I'll correct.

Reply:
Corrected. I left the sync failure event on the idea that we only ever want to be notified when synchronization fails, as it runs every 2 minutes and could get noisy. Let me know if you feel strongly otherwise.